packages/file-explorer/src/NewItemModal.tsx

```diff
@@ -332,10 +345,13 @@ class NewItemModal extends PureComponent<NewItemModalProps, NewItemModalState> {
         .catch(e => {
           // Don't allow using existing names for folders
           // For files, prompt if they want to overwrite existing file
-          if (!(e instanceof FileExistsError) || type === 'directory') {
-            throw e;
+          if (e instanceof FileExistsError) {
+            if (type !== 'directory' && e.info.type !== 'directory') {
+              this.setState({ showOverwriteModal: true });
+              return;
+            }
           }
-          this.setState({ showOverwriteModal: true });
+          throw e;
```

Comment:
I'm getting an error `Cannot overwrite existing directory` when trying to save a file with the same name as an existing directory. Should be allowed, I think.

Reply:
As discussed in meeting, it should not be allowed to rename a file to overwrite a directory.